### DISCUSSION TITLE
OTA over Ethernet

### DISCUSSION THREAD

**Discussion by MarcinKlima** [earlephilhower/arduino-pico#2579] [General]
Hi, trying  to run ota on wiznet Ethernet hat (as w5100), run clean project, with working wiznet module, got ip address (network is working)and add OTA from example BasicOTA, the rp2040 freeze on:  ArduinoOTA.begin();  wiznet lib is from Arduino mean "Ethernet" Any idea? :)

**Comment by JAndrassy** [comment]
to use BasicOTA you have to use Ethernet for this platform
https://github.com/earlephilhower/arduino-pico/blob/master/libraries/lwIP_Ethernet/examples/EthernetLibCompatible/EthernetLibCompatible.ino

**Reply by MarcinKlima** [reply]
Yes, **now work perfect**, all features from network like esp32 work well ;) ota work now like a harm.  Not work because: 3 problems, not possible to add to project "EthernetCompat.h"  because, on platformio: platform_packages = maxgerhardt/framework-arduinopico@https://github.com/earlephilhower/arduino-pico.git and  after remove it im don`t update platform, after pio pkg update -g -p https://github.com/maxgerhardt/platform-raspberrypi.git. After cleaning all and fresh install of the platform, all lib`s and functions from earlephilhower working good, thank You!

**Comment by maxgerhardt** [comment]
This topic already has the same question, and solution: https://github.com/earlephilhower/arduino-pico/discussions/2556. I've also imported that reference project into the platform now at https://github.com/maxgerhardt/platform-raspberrypi/tree/develop/examples/arduino-ethernet-ota.
